runtime/drivers/duckdb/model_executor_objectstore_self.go

```diff
@@ -0,0 +1,166 @@
+	if p.Path == "" {
+		return fmt.Errorf("missing property `path`")
+	}
+	return nil
+}
+
+type objectStoreToSelfExecutor struct {
+	c *connection
+}
+
+var _ drivers.ModelExecutor = &objectStoreToSelfExecutor{}
+
+func (e *objectStoreToSelfExecutor) Concurrency(desired int) (int, bool) {
+	if desired > 1 {
+		return 0, false
+	}
+	return 1, true
+}
+
+func (e *objectStoreToSelfExecutor) Execute(ctx context.Context, opts *drivers.ModelExecuteOptions) (*drivers.ModelResult, error) {
+	// Build the model executor options with updated input properties
+	clone := *opts
+	newInputProps, err := e.modelInputProperties(opts.ModelName, opts.InputConnector, opts.InputHandle, opts.InputProperties)
+	if err != nil {
+		return nil, err
+	}
+	clone.InputProperties = newInputProps
+	newOpts := &clone
+
+	// execute
+	executor := &selfToSelfExecutor{c: e.c}
+	return executor.Execute(ctx, newOpts)
+}
+
+func (e *objectStoreToSelfExecutor) modelInputProperties(model, inputConnector string, inputHandle drivers.Handle, inputProps map[string]any) (map[string]any, error) {
+	parsed := &s3InputProps{}
+	if err := mapstructure.WeakDecode(inputProps, parsed); err != nil {
+		return nil, fmt.Errorf("failed to parse input properties: %w", err)
+	}
+	if err := parsed.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid input properties: %w", err)
+	}
+
+	m := &ModelInputProperties{}
+	var format string
+	if parsed.Format != "" {
+		format = fmt.Sprintf(".%s", parsed.Format)
+	} else {
+		format = fileutil.FullExt(parsed.Path)
+	}
+
+	config := inputHandle.Config()
+	// config properties can also be set as input properties
+	maps.Copy(config, inputProps)
+
+	// Generate secret SQL to access the service and set as pre_exec_query
+	safeSecretName := safeName(fmt.Sprintf("%s__%s__secret", model, inputConnector))
+	switch inputHandle.Driver() {
+	case "s3":
+		s3Config := &s3.ConfigProperties{}
+		err := mapstructure.WeakDecode(config, s3Config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse s3 config properties: %w", err)
+		}
+		var sb strings.Builder
+		sb.WriteString("CREATE OR REPLACE TEMPORARY SECRET ")
+		sb.WriteString(safeSecretName)
+		sb.WriteString(" (TYPE S3")
+		if s3Config.AllowHostAccess {
+			sb.WriteString(", PROVIDER CREDENTIAL_CHAIN")
+		}
+		if s3Config.AccessKeyID != "" {
+			fmt.Fprintf(&sb, ", KEY_ID %s, SECRET %s", safeSQLString(s3Config.AccessKeyID), safeSQLString(s3Config.SecretAccessKey))
+		}
+		if s3Config.SessionToken != "" {
+			fmt.Fprintf(&sb, ", SESSION_TOKEN %s", safeSQLString(s3Config.SessionToken))
+		}
+		if s3Config.Endpoint != "" {
+			sb.WriteString(", ENDPOINT ")
+			sb.WriteString(safeSQLString(s3Config.Endpoint))
+		}
+		if s3Config.Region != "" {
+			sb.WriteString(", REGION ")
+			sb.WriteString(safeSQLString(s3Config.Region))
+		}
+		sb.WriteRune(')')
+		m.PreExec = sb.String()
+	case "gcs":
+		// GCS works via S3 compatibility mode
+		s3Config := &s3.ConfigProperties{}
```

Comment:
See other comment, maybe we should use `gcs.ConfigProperties` here and use Google-related property names?